saml-core/src/test/java/org/keycloak/saml/processing/core/parsers/saml/SAMLParserTest.java

```diff
@@ -216,4 +223,102 @@ public void testEmptyAttributeValueLast() throws Exception {
+        if ("saml20-assertion-anytype-attribute-value.xml".equals(fileName)) {
+            assertTrue("There has to be 3 attributes", attributeStatementType.getAttributes().size() == 3);
+            for (AttributeStatementType.ASTChoiceType choiceType: attributeStatementType.getAttributes()) {
+                AttributeType attr = choiceType.getAttribute();
+                String attrName = attr.getName();
+                String attrValueStatement = "unexpected value of attribute " + attrName + " of " + fileName;
+                String attrTypeStatement = "unexpected type of attribute " + attrName + " of " + fileName;
+                // test selected attributes
+                if (attrName.equals("attr:type:string")) {
+                    assertTrue(attrValueStatement, attr.getAttributeValue().get(0).equals("CITIZEN"));
+                } else if (attrName.equals("attr:notype:string")) {
+                    assertTrue(attrTypeStatement, attr.getAttributeValue().get(0) instanceof String);
+                    String value = (String)attr.getAttributeValue().get(0);
+                    assertTrue(attrValueStatement, value.equals("CITIZEN"));
+                } else if (attrName.equals("attr:notype:element")) {
+                    assertTrue(attrTypeStatement, attr.getAttributeValue().get(0) instanceof String);
+                    String value = (String)attr.getAttributeValue().get(0);
+                    assertTrue(attrValueStatement, value.contains("hospitaal x"));
+                    value = (String)attr.getAttributeValue().get(1);
+                    assertTrue(attrValueStatement, value.contains("hopital x"));
+                }
+            }
+        } else if ("saml20-assertion-example.xml".equals(fileName)) {
+            assertTrue("There has to be 8 attributes", attributeStatementType.getAttributes().size() == 8);
+            for (AttributeStatementType.ASTChoiceType choiceType: attributeStatementType.getAttributes()) {
+                AttributeType attr = choiceType.getAttribute();
+                String attrName = attr.getName();
+                String attrValueStatement = "unexpected value of attribute " + attrName + " of " + fileName;
+                String attrTypeStatement = "unexpected type of attribute " + attrName + " of " + fileName;
+                // test selected attributes
+                if (attrName.equals("portal_id")) {
+                    assertTrue(attrValueStatement, attr.getAttributeValue().get(0).equals("060D00000000SHZ"));
+                } else if (attrName.equals("organization_id")) {
+                    assertTrue(attrTypeStatement, attr.getAttributeValue().get(0) instanceof String);
+                    String value = (String)attr.getAttributeValue().get(0);
+                    assertTrue(attrValueStatement, value.contains("<n3:stuff xmlns:n3=\"ftp://example.org\">00DD0000000F7L5</n3:stuff>"));
```

Comment:
Similarly, to get better output, use `Assert.assertThat(attrValueStatement, value, containsString("..."));`